src/main/java/com/heapwhisperer/metrics/jvm/NonAccumulatingGarbageCollectorMetricSet.java

```diff
@@ -35,37 +37,47 @@
     private Map<String, Long> nonAccumulatingValues;
     private GarbageCollectorMetricSet garbageCollectorMetricSet;
     private ScheduledExecutorService scheduledExecutorService;
-    private long interval;
+    private long intervalMs;
+    private String intervalInfix;
 
     /**
      * Constructor does not take an executor service, instead deferring
      * to the default scheduled executor service provided by this class.
      *
      * @param garbageCollectorMetricSet a metric set that collects counts and times of garbage collections
-     * @param interval the time interval over which to calculate non-accumulating gauge readings
+     * @param intervalMs the time interval in milliseconds over which to calculate non-accumulating gauge readings
      *                 for all the gauges in {@code garbageCollectorMetricSet}
      */
     public NonAccumulatingGarbageCollectorMetricSet(
-            GarbageCollectorMetricSet garbageCollectorMetricSet, long interval) {
-        this(garbageCollectorMetricSet, interval, null);
+            GarbageCollectorMetricSet garbageCollectorMetricSet, long intervalMs) {
+        this(garbageCollectorMetricSet, intervalMs, null);
     }
 
     /**
      * Constructor sets up the scheduled executor service that runs a background task to
      * calculate non-accumulating gauge readings at periodic intervals.
      *
      * @param garbageCollectorMetricSet a metric set that collects counts and times of garbage collections
-     * @param interval the time interval over which to calculate non-accumulating gauge readings
+     * @param intervalMs the time interval in milliseconds over which to calculate non-accumulating gauge readings
      *                 for all the gauges in {@code garbageCollectorMetricSet}
      * @param scheduledExecutorService scheduled executor service that runs the task to calculate
      *                                 non-accumulating gauge readings at a frequency determined by
-     *                                 {@code interval}.
+     *                                 {@code intervalMs}.
      */
     public NonAccumulatingGarbageCollectorMetricSet(
-            GarbageCollectorMetricSet garbageCollectorMetricSet, long interval,
+            GarbageCollectorMetricSet garbageCollectorMetricSet, long intervalMs,
             ScheduledExecutorService scheduledExecutorService) {
         this.garbageCollectorMetricSet = garbageCollectorMetricSet;
-        this.interval = interval;
+        this.intervalMs = intervalMs;
+        if (intervalMs >= 60000) {
+            long minutes = TimeUnit.MILLISECONDS.toMinutes(intervalMs);
+            this.intervalInfix = "last-" + minutes + "-minute" + (minutes > 1 ? "s" : "");
```

Comment:
If `intervalMs` is always in milliSeconds why not just keep `intervalInfix` always in milliseconds?